src/plugins/intel_cpu/src/nodes/if.cpp

```diff
@@ -17,31 +19,43 @@ namespace ov {
 namespace intel_cpu {
 namespace node {
 
+inline size_t getElementNum(const MemoryPtr& srcMemPtr) {
+    auto srcDims = srcMemPtr->getStaticDims();
+    return std::accumulate(srcDims.begin(), srcDims.end(), 1, std::multiplies<size_t>());
+}
+
 If::PortMapHelper::PortMapHelper(const MemoryPtr &from, const std::deque<MemoryPtr>& to,
                                            const dnnl::engine& eng) : srcMemPtr(from), dstMemPtrs(to) {
     size = 0;
     if (srcMemPtr->getDesc().isDefined())
-        size = srcMemPtr->getSize();
+        size = getElementNum(srcMemPtr);
 }
 
 void If::PortMapHelper::execute(dnnl::stream& strm) {
     // if output shapes are changed,
     // after subgraph inference we should redefine out memory of 'If'
     redefineTo();
 
-    cpu_memcpy(dstMemPtrs.front()->getData(), srcMemPtr->getData(), size);
+    cpu_convert(srcMemPtr->getData(),
+                dstMemPtrs.front()->getData(),
+                srcMemPtr->getDesc().getPrecision(),
+                dstMemPtrs.front()->getDesc().getPrecision(),
+                size);
 }
 
 void If::PortMapHelper::redefineTo() {
     const auto &currDesc = dstMemPtrs.front()->getDesc();
     if (currDesc.getShape().isDynamic() || currDesc.getShape().getStaticDims() != srcMemPtr->getStaticDims()) {
         // TODO : check the entire dstMemPtrs usage considering the proper memory sharing
-        auto memDesc = srcMemPtr->getDescPtr();
+        auto newShape = srcMemPtr->getDesc().getShape().getDims();
         for (size_t j = 0; j < dstMemPtrs.size(); j++) {
-            dstMemPtrs[j]->redefineDesc(memDesc);
+            // Only the shape is updated, the memory type remains unchanged
+            auto memDesc = dstMemPtrs[j]->getDescPtr();
+            const auto desc = std::make_shared<CpuBlockedMemoryDesc>(memDesc->getPrecision(), Shape(newShape));
```

Comment:
I'm not sure that new `CpuBlockedMemoryDesc` creation  for `If` op output is good idea since it may has not default parameters (order, offsets, smth else) in general. Because of that I'd suggest to call `getBaseMemDescAtOutputPort(map_rule.from)->cloneWithNewDims(newShape)`
What do you think?

Reply:
Good idea, but it is not easy to get `map_rule`, because we don't know it is **then node** or **else node**, maybe we can add some parameters to deal with it.

I just backup `dstMemPtrs` in constructor, and then we just use backup's` dstMemPtrs` to `redefineDesc`, is it OK? @a-sidorova